commands/bake.go

```diff
@@ -201,7 +197,26 @@ func runBake(dockerCli command.Cli, targets []string, in bakeOptions, cFlags com
 		return nil
 	}
 
-	resp, err := build.Build(ctx, nodes, bo, dockerutil.NewClient(dockerCli), confutil.ConfigDir(dockerCli), printer)
+	var lsg *localstate.StateGroup
+	if len(bo) > 1 {
+		var refs []string
+		for _, b := range bo {
+			refs = append(refs, b.Ref)
+		}
+		dt, err := json.Marshal(def)
+		if err != nil {
+			return err
+		}
+		lsg = &localstate.StateGroup{
+			Ref:        identity.NewID(),
+			Definition: dt,
+			Targets:    targets,
+			Inputs:     overrides,
+			Refs:       refs,
+		}
```

Comment:
Why not just write the file in here and not pass `lsg`?

To get the `groupID` you can just add `GroupRef` in the build options. The regular `Ref` can also be set in here if instead of other places.

Reply:
To have the local refs being saved and then the local group in the same place in `build.Build`. Otherwise we could potentially have dangling group refs.

Reply:
There is always a chance that something is dangling in this storage for panic. If you write ref first, then it can have a groupRef that does not exist; that is potentially a worse case because that could really cause a lookup error.

I'm just trying to reduce the area impacted by the PR. I think setting `GroupRef` value to build options is fine but would like to avoid passing big "group state" structs with their own lifecycle through different functions.